bean-utils-library/src/test/java/com/hotels/beans/BeanUtilsTest.java

```diff
@@ -109,6 +113,26 @@ public void testGetTransformerFunctionWorksProperly(final String testCaseDescrip
                 .forEach(i -> assertThat(actual.get(i), sameBeanAs(fromFooSimpleList.get(i))));
     }
 
+    /**
+     * Test that the transformer function is applied earlier than the default value
+     */
+    @Test
+    public void testTransformerFunctionHasHigherPriorityThanDefaultValue() {
```

Comment:
as you are transforming an Immutable object, could you kindly move this test into the test class: `ImmutableObjectTransformationTest`? 
thanks